Cython/Compiler/ExprNodes.py

```diff
@@ -7137,6 +7151,9 @@ def analyse_attribute(self, env, obj_type = None):
         if obj_type.has_attributes:
             if obj_type.attributes_known():
                 entry = obj_type.scope.lookup_here(self.attribute)
+                if (obj_type.is_cpp_class and (self.obj.is_name or self.obj.is_attribute) and
+                        self.obj.entry and self.obj.entry.is_cpp_optional):
+                    self.op = "->"
```

Comment:
Is this the right place for this code? Not further up, where we set `self.op` in other cases?
I'm mostly asking because we just looked up the attribute's entry here, but this code then doesn't make use of it. That seems a bit suspicious.